### ISSUE TITLE
Return list of missing required fields

### ISSUE THREAD

**Issue by jamesaimonetti** [for-GET/jesse#29]
When validating an object against a schema with multiple fields in the `required` array (draft4), it would be great to get back either:

1. `{missing_required_property, [<<"foo">>, <<"bar">>]}`
2. Multiple 'data_invalid' tuples with each tuple having `{missing_required_property, Key}`

It looks straightforward enough in the code, something like (untested):

```erlang
check_required(Value, [_|_]=Required, State) ->
    {_, Missing} = lists:partition(fun(Property) -> get_value(PropertyName, Value) =/= ?not_found end, Required),
    case Missing of
        [] -> State;
        MissingKeys -> %% do appropriate error return
    end;
```
Thoughts? Happy to submit the PR if this is a worthwhile change.

**Comment by andreineculau** [comment]
Sounds like a good idea! Welcome to send a PR

PS: watch out for support of `allowed_errors`.

**Comment by andreineculau** [comment]
@jamesaimonetti no hurry, but FYI I intend to make a release before xmas

**Comment by marutha** [comment]
+1 for this

**Comment by andreineculau** [comment]
will be fixed by #35
